plugins/module_utils/network/sonic/config/interfaces/interfaces.py

```diff
@@ -282,26 +302,226 @@ def _state_deleted(self, want, have, diff):
+                if request:
+                    requests.append(request)
+                    new_mer_cmd = True
+
+                if new_mer_cmd:
+                    mer_commands.append(conf)
+
+        return del_commands, mer_commands, requests
+
+    def build_delete_description_request(self, conf, have_conf):
+        intf_name = conf['name']
+        request = dict()
+        method = DELETE
+
+        c_des = conf.get('description', None)
+        h_des = have_conf.get('description', None)
+        if c_des and h_des and h_des != '':
+            config_url = (url + '/config/description') % quote(intf_name, safe='')
+            request = {"path": config_url, "method": method}
+
+            conf['description'] = ''
+
+        return request
+
+    def build_delete_enabled_request(self, conf, have_conf):
+        intf_name = conf['name']
+        request = dict()
+        method = DELETE
+
+        c_ena = conf.get('enabled', None)
+        h_ena = have_conf.get('enabled', None)
+        if c_ena is not None and h_ena is not None and h_ena:
+            config_url = (url + '/config/enabled') % quote(intf_name, safe='')
+            request = {"path": config_url, "method": method}
+
+            conf['enabled'] = False
+
+        return request
+
+    def build_delete_mtu_request(self, conf, have_conf):
+        intf_name = conf['name']
+        request = dict()
+        method = DELETE
+
+        if not intf_name.startswith('Loopback'):
+            c_mtu = conf.get('mtu', None)
+            h_mtu = have_conf.get('mtu', None)
+            if c_mtu and h_mtu and h_mtu != 9100:
+                config_url = (url + '/config/mtu') % quote(intf_name, safe='')
+                request = {"path": config_url, "method": method}
+
+                conf['mtu'] = 9100
+
+        return request
+
+    def build_delete_fec_request(self, conf, have_conf):
+        intf_name = conf['name']
+        request = dict()
+        method = PATCH
+
+        if intf_name.startswith('Ethernet'):
+            c_fec = conf.get('fec', None)
+            h_fec = have_conf.get('fec', None)
+            if c_fec and h_fec and h_fec != 'FEC_DISABLED':
+                fec_url = '/openconfig-if-ethernet-ext2:port-fec'
+                eth_url = '/openconfig-if-ethernet:ethernet/config'
+                config_url = (url + eth_url + fec_url) % quote(intf_name, safe='')
+                payload = {'openconfig-if-ethernet-ext2:port-fec': 'FEC_DISABLED'}
+                request = {"path": config_url, "method": method, 'data': payload}
+
+                conf['fec'] = 'FEC_DISABLED'
+
+        return request
+
+    def build_delete_speed_request(self, conf, have_conf):
+        intf_name = conf['name']
+        request = dict()
+        method = DELETE
+
+        if intf_name.startswith('Ethernet'):
+            c_spd = conf.get('speed', None)
+            h_spd = have_conf.get('speed', None)
+            if c_spd and h_spd:
```

Comment:
Usually, deletion of an attribute occurs only if the specified "deletion" value for the attribute matches the current value of that attribute.

Is there some reason this line doesn't check for that usual requirement with something like the following?

```
if c_spd and h_spd and c_spd == h_spd:
```